### ISSUE TITLE
Make `kedro ipython/jupyter` commands not use the project's `.ipython/` dir

### ISSUE THREAD

**Issue by idanov** [kedro-org/kedro#1338]
Currently Kedro is using a project specific `.ipython/` in order to preload the Kedro IPython extension when running `kedro ipython` and `kedro jupyter notebook/lab`. This is a hack we've been using to simplify the user experience with Jupyter and IPython, but it's not a widely used practice and often causes troubles.

One of the most common problems with this approach is that the user needs to run the Jupyter instances themselves, which is not applicable in environments with managed notebook instances like SageMaker and Databricks.

In order to pave the way to a state where there's very little distinction between local and managed Jupyter instances, we need to modify our commands to do as follows:

- `kedro ipython` should be changed to simply be an alias of `ipython --ext="kedro.extras.extensions.ipython"`
- `kedro jupyter lab/notebook` should take advantage of Jupyter Kernels
    - it should install a new Jupyter Kernel (if it doesn't exist) named after the value of `from kedro.framework.project import PACKAGE_NAME` with display name `Kedro (<PACKAGE_NAME>)`
    - the new kernel should load the `kedro.extras.extensions.ipython` IPython extension on start
    - the new kernel should use the Kedro icon
    - it should run Jupyter by calling `jupyter notebook --MultiKernelManager.default_kernel_name=<PACKAGE_NAME>`
- make `kedro ipython` and `kedro jupyter` print out a nice error message if we don't have `jupyter_client`, `jupyter` or `ipython` installed (and remove the dependency of Kedro on `jupyter_client`)
- remove all `.ipython/` folders in our official starters
- update all our docs mentioning `.ipython/`

**Comment by WaylonWalker** [comment]
Will this then run my profile_default.  Once primary reason I personally don't often use `kedro ipython` is because it uses the projects profile, rather than my custom profile.  Based on the alias it looks like it would, and this would be a HUGE improvement for me.

**Comment by idanov** [comment]
@WaylonWalker yes, now you will have access to your own custom `profile_default`. Behind the scenes `kedro ipython` will only run `ipython --ext="kedro.extras.extensions.ipython"`, which will only add the extension and will pick up your default `profile_default` on your machine.
